app/models/draft_vaccination_record.rb

```diff
@@ -131,7 +131,11 @@ def already_had?
   alias_method :administered, :administered?
 
   def protocol
-    :pgd
+    if session.psd_enabled? && @current_user.is_healthcare_assistant?
```

Comment:
Are these two conditions sufficient? We might not need to check if PSD is added here because the HCA wouldn't have been allowed to see the record vaccination section on the patient page if there's no PSD added.

Reply:
I think we need to do a more thorough check here, particular in relation to the national protocol (#4409) because I there could be a scenario where a patient has a PSD for nasal flu, the national protocol is enabled, and therefore the HCA decides to vaccinate using the injection, which would mean the protocol is `national`, not `psd`.

Reply:
Updated the logic including a feature scenario to cover this edge case. Is the logic ok?

Reply:
If both were enabled, which protocol gets used?